### PR TITLE
Referral

### PR DESCRIPTION
**Pull request by astrocket**

### CODE REVIEW

File: src/Routes.js [hunk 1]
```diff
@@ -114,10 +114,11 @@ class Right extends Component {
     const AuthorList = isEmpty(this.props.searchTerm) ? HuntedListByAuthor : Search;
     const TopList = isEmpty(this.props.searchTerm) ? HallOfFame : Search;
     const TagRight =  isEmpty(this.props.searchTerm) ? TagList : Search;
+    const params = new URLSearchParams(this.props.location.search);
 
     return (
       <div className="panel-right">
-        {this.props.location.search && <Redirect to="/" /> /* Authentication redirection */ }
+        {params.get('access_token') && params.get('username') && <Redirect to="/" />} {/* Authentication redirection */}
```
Comment (sydneyitguy) [inline src/Routes.js]: I've also updated this on the master branch. Please pull this code from 
https://github.com/Steemhunt/web/blob/master/src/Routes.js#L118-L125

**Comment by sydneyitguy** [comment]
From the discussion about UX disadvantages,
We may need to remove the popup and use scroll-detection for preloader / bot filtering.

Additional work:
1. Show SH logo instead of back button when mobile / direct access on product page
2. Show "What is Steemhunt" banners on the top of the page, and link to "About" page on click (@cyh76507707 I guess we need a design for this)

Sorry for the change after work.

**Comment by cyh76507707** [comment]
image


After the user click the button, please redirect to the about page. 

Also, when the user check go to the about page, I think we need to show "go to today's chart" message on the back button, and the back button should lead them to the main page.

image

**Comment by sydneyitguy** [comment]
I've finished the pre-rendering stuff now, so please do the og:image stuff using Helmet and check social services can render the site properly.

**Comment by sydneyitguy** [comment]
Meta tags and og:image stuff is done. Please test it on some social sites .

**Comment by cyh76507707** [comment]
Let's re-organise share buttons.

1. Move the resteem button on bottom right like the image below.
screenshot 2018-09-06 13 43 23

2. Create new share button, and once user "click" it, show the social share buttons (Facebook, Twitter, Pinterest, Linkedin). Please see the codepen below, and use the source for the animation effect.
https://codepen.io/yancy/pen/qyMdod

3. Put this title on the share button --> `Share and get HUNT tokens`
- Button location: the current "resteem" button
- Border-color: #FC6F6F
- No bg color
- When hoover: reverse the color
- When clicked: make the social buttons appeared

4. For the social icons, please use Fontastic

**Comment by astrocket** [comment]
image

Currently layout is being scrolled from inner div. either from routes-left or routes-right. (acting separately)

image

Since we cannot predict which page user will be referaled into, we either have to put top banner into both side or fix the banner on top of the screen.

Any better way?

**Comment by astrocket** [comment]
@sydneyitguy It's already in App.js however we cannot control the scroll action since actual scroll of the device is handled from the inner div of any page from routes-left / routes-left. 

We either have to fix all the layout of pages  to let top banner be get scrolled up as user scrolls.
Or just fix the top banner on top. Fixed when user scrolls. No modification of default inner div scroll layout.

**Comment by sydneyitguy** [comment]
@astrocket oh.. that was the issue.
I guess we can just add a close button on the top banner (just "X" button on the right side).

**Comment by astrocket** [comment]
Ok that's what I thought too. got it.

**Comment by astrocket** [comment]
@cyh76507707 I need confirmation for the banner style.

image

**Comment by cyh76507707** [comment]
**[Content in the modal screen]**

<div class="ant-modal-header"><div class="ant-modal-title" id="rcDialogTitle0">Share the hunt and get free HUNT tokens!</div></div>
<div class="ant-modal-body"><p>90,000 tokens per day are distributed for people who share hunting posts to their social media channels including Facebook, Twitter, Pinterest, and Linkedin.</p><br>
<p>HUNT tokens will be assigned based on the share of the daily total traffics generated from all the shared posts via the social channels. For example, if today’s total visitors from the social shared posts are 10,000, and your shared posts has generated 100 visitors, your share will be  1%. In this case, you will get 900 HUNT tokens (90,000*1%).</p><br>
<p>Maximum tokens per person per day is 1,000 tokens.</p></div>

File: src/features/App/Referral.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  scrollDetector(e) {
+    if ((e.srcElement.scrollTop > 300) && !this.state.reported) {
+      this.postReferral()
+    }
+  }
+
+  postReferral() {
+    this.setState({ reported: true }, () => {
+      api.post('/referral', {
```
Comment (sydneyitguy) [inline src/features/App/Referral.js]: I have changed the parameters on the API

1. `type` has been removed
2. `referrer` has been added (it should be `document.referrer`)

File: src/features/App/Referral.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom';
+import api from 'utils/api';
+import { Icon } from 'antd';
+
+class Referral extends Component {
```
Comment (sydneyitguy) [inline src/features/App/Referral.js]: 👍

File: src/features/Post/components/PostView.js [hunk 4]
```diff
@@ -212,6 +210,8 @@ class PostView extends Component {
       </Button>
     );
 
+    const shareUrl = window.location.href + (me ? `%3Fref=${me}%26` : '%3F');
```
Comment (sydneyitguy) [inline src/features/Post/components/PostView.js]: this seems unnecessary here, looks like it's moved to `ShareButton.js`

File: src/features/Post/components/ShareButton.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+  render() {
+    const { post, me } = this.props;
+    const shareUrl = window.location.href + (me ? `%3Fref=${me}%26` : '%3F');
+
+    const content = (
+      <div className="social-shares">
+        <Tooltip title="Share on Facebook">
+          <a
+            href={'https://www.facebook.com/sharer.php?u=' + shareUrl + 'type=1'}
```
Comment (sydneyitguy) [inline src/features/Post/components/ShareButton.js]: we don't need `type` param anymore

File: src/styles/_common.scss [hunk 6]
```diff
@@ -139,6 +139,34 @@ hr {
...
+    z-index: 320;
+    width: 100%;
+    display: flex;
+    justify-content: space-between;
+    padding: 0px 20px;
+    align-items: center;
+    background: #333;
+    button {
+      display: flex;
+      padding: 0px;
```
Comment (sydneyitguy) [inline src/styles/_common.scss]: Please use `0` over `0px` (convention because units don't matter if 0)

**Pull request merged**